tests/settings_tests/tests.py

```diff
@@ -441,3 +441,132 @@ def test_tuple_settings(self):
             finally:
                 del sys.modules['fake_settings_module']
                 delattr(settings_module, setting)
+
+
+class SettingChangeEnterException(Exception):
+    pass
+
+
+class SettingChangeExitException(Exception):
+    pass
+
+
+class OverrideSettingsIsolationOnExceptionTest(SimpleTestCase):
+    """
+    Test that ``override_settings`` context manager restores settings
+    when at least one of the receivers of "setting_changed" signal fails.
+
+    Do this with three cases of failure:
+
+    1) receiver fails when the signal is sent while both entering and exiting the context manager,
+    2) receiver fails on enter,
+    3) receiver fails on exit.
+
+    Make sure that in each case ALL receivers are called when exiting the context manager.
+    """
+
+    def setUp(self):
+        signals.setting_changed.connect(self.receiver)
+        self.addCleanup(signals.setting_changed.disconnect, self.receiver)
+
+        # Create a spy that will be connected to `setting_changed` signal
+        # and executed AFTER `self.receiver`.
+        self.spy_receiver = mock.Mock()
+        signals.setting_changed.connect(self.spy_receiver)
+        self.addCleanup(signals.setting_changed.disconnect, self.spy_receiver)
+
+    def receiver(self, **kwargs):
+        """
+        A receiver for demonstration purposes.
+        It fails while certain settings are being changed.
+
+        - BUG_29467_BOTH raises an error while receiving the signal
```

Comment:
I think something like `SETTING_BOTH` will be fine. No need to memorialize the bug number.

Reply:
Right, I wanted to come up with a name that's unique enough so no other test (that e.g. forgot to do proper clean ups) would cause name collisions and thus test failure. But I'll make the change.